### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by mviswanathsai**
## Description

Documentation on deploying the local code in a local cluster seems like a good fit in [CONTRIBUTING.md](https://github.com/prometheus-operator/prometheus-operator/blob/main/CONTRIBUTING.md). However, it has not been removed from [TESTING.md](https://github.com/prometheus-operator/prometheus-operator/blob/main/TESTING.md) where it was originally part of, because it seems relevant there too.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```
## Related 
Slack discussion: https://kubernetes.slack.com/archives/C01B03QCSMN/p1709908179897869.

### CODE REVIEW

**Comment by nicolastakashi** [comment]
Thanks for handle this @mviswanathsai, I just left a small comment non-blocking.
I think you also need to run `make --always-make format generate`

**Comment by mviswanathsai** [comment]
Thank you for the review! I have made the change you suggested and also ran the make command.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/prometheus-operator/prometheus-operator
 
-go 1.22
```
Comment (nicolastakashi) [inline go.mod:3]: could you please revert it?

Comment (mviswanathsai) [reply go.mod:3]: Sure. I was getting errors with resolving this in my local machine, hence the change. I will revert it.

**Comment by mviswanathsai** [comment]
Seems like commits from other people have made its way here.

**Comment by mviswanathsai** [comment]
This PR has been moved to #6388.

**Pull request closed without merging**